mysticeti-core/src/validator.rs

```diff
@@ -125,12 +91,118 @@ impl Validator {
+        tokio::sync::mpsc::Sender<(Vec<u8>, tokio::sync::oneshot::Sender<()>)>,
+    )> {
+        let (metrics, reporter, metrics_handle) =
+            Self::init_metrics(authority, committee.clone(), parameters, Some(registry))?;
+        let (recovered, wal_writer) =
+            Self::init_storage(authority, committee.clone(), &config, metrics.clone());
+        reporter.start();
+        let (block_handler, tx_sender) = SimpleBlockHandler::new();
+
+        let validator = Validator::start_internal(
+            authority,
+            committee,
+            parameters,
+            config,
+            signer,
+            metrics,
+            metrics_handle,
+            recovered,
+            wal_writer,
+            block_handler,
+        )
+        .await?;
+
+        Ok((validator, tx_sender))
+    }
+}
+
+impl<B: BlockHandler + 'static> Validator<B> {
+    pub async fn start(
+        authority: AuthorityIndex,
+        committee: Arc<Committee>,
+        parameters: &Parameters,
+        config: PrivateConfig,
+        signer: Signer,
+        registry: Option<Registry>,
+        block_handler: B,
+    ) -> Result<Self> {
+        let (metrics, reporter, metrics_handle) =
+            Self::init_metrics(authority, committee.clone(), parameters, registry)?;
+        let (recovered, wal_writer) =
+            Self::init_storage(authority, committee.clone(), &config, metrics.clone());
+        reporter.start();
+
+        Self::start_internal(
+            authority,
+            committee,
+            parameters,
+            config,
+            signer,
+            metrics,
+            metrics_handle,
+            recovered,
+            wal_writer,
+            block_handler,
+        )
+        .await
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    async fn start_internal(
+        authority: AuthorityIndex,
+        committee: Arc<Committee>,
+        parameters: &Parameters,
+        config: PrivateConfig,
+        signer: Signer,
+        metrics: Arc<Metrics>,
+        metrics_handle: MetricsHandle,
+        recovered: RecoveredState,
+        wal_writer: WalWriter,
+        block_handler: B,
+    ) -> Result<Self> {
+        let network_address = parameters
+            .network_address(authority)
+            .ok_or(eyre!("No network address for authority {authority}"))
+            .wrap_err("Unknown authority")?;
+        let mut binding_network_address = network_address;
+        binding_network_address.set_ip(IpAddr::V4(Ipv4Addr::UNSPECIFIED));
+
+        // Boot the validator node.
         let committed_transaction_log =
             TransactionLog::start(config.storage().committed_transactions_log())
                 .expect("Failed to open committed transaction log for write");
         let commit_handler = TestCommitObserver::new_with_handler(
             committee.clone(),
-            block_handler.transaction_time.clone(),
+            block_handler.transaction_time(),
```

Comment:
We will need to change this too and use commit observer from here: https://github.com/MystenLabs/mysticeti/pull/42

Note that things like `committed_transaction_log` are also benchmark-specific and won't be used in sui (the sui block handler in the PR above does not take committed_transaction_log parameter)